### PR TITLE
Add commands for Sources and Executors

### PR DESCRIPTION
**Pull request by josefkarasek**
## Description

Changes proposed in this pull request:

- Add commands for listing Executors bound to a channel
- Add commands for listing Sources bound to a channel
- Remove command `list commands`

### Testing

1. Create `/tmp/test-values.yaml`:

```yaml
plugins:
  cacheDir: "/tmp/plugins"
  repositories:
    botkube:
      url: http://host.k3d.internal:3000/botkube.yaml

executors:
  'plugin-based':
    botkube/helm:
      enabled: true
      config: {}
  'kubectl-allow-all':
    kubectl:
      enabled: true
      namespaces:
        include:
          - ".*"
      commands:
        verbs: [ "get" ]
        resources: [ "deployments" ]
  'kubectl-not-bound-to-any-channel':
    kubectl:
      enabled: true
      namespaces:
        include:
          - ".*"
      commands:
        verbs: [ "port-forward" ]
        resources: [ "deployments" ]

sources:
  'plugin-based':
    displayName: "Events based on plugin"
    botkube/helm:
      enabled: true
      config: {}
  'k8s-pod-create-events':
    displayName: "k8s events for CREATE pod"
    kubernetes:
      namespaces:
        include:
          - botkube
          - test
      events:
        - create
      resources:
        - name: v1/pods

communications:
  'default-group':
    socketSlack:
      appToken: xapp-***
      botToken: xoxb-***
      channels:
        default:
          bindings:
            executors:
            - plugin-based
            - kubectl-allow-all
            - kubectl-not-bound-to-any-channel
            sources:
            - plugin-based
            - k8s-pod-create-events
          name: botkube
      enabled: true
      notification:
        type: short

rbac:
  create: true
  rules:
    - apiGroups: ["*"]
      resources: ["*"]
      verbs: ["get", "watch", "list", "create", "delete"]
```

2. Start k3d cluster

```bash
k3d cluster create labs --image=rancher/k3s:v1.25.0-k3s1
```
3. Build plugins

```bash
make build-plugins
```
4. Start plugin server:

```bash
env PLUGIN_SERVER_HOST=http://host.k3d.internal go run test/helpers/plugin_server.go
```
5. Install Botkube

```bash
helm upgrade botkube --install --namespace botkube ./helm/botkube --wait --create-namespace \
  -f /tmp/test-values.yaml \
  --set image.repository=kubeshop/pr/botkube \
  --set image.tag=905-PR \
  --set image.pullPolicy="Always"
```

### Steps

1. List commands is no longer supported

```
@Botkube list commands
```

2. New cmd `list source` should show enabled sources

```
@Botkube list src

botkube/helm
k8s-pod-create-events
```

3. New cmd `list executors`

```
@Botkube list executors

botkube/helm
kubectl-allow-all
kubectl-not-bound-to-any-channel
```

## Related issue(s)

Resolves https://github.com/kubeshop/botkube/issues/891

### CODE REVIEW

File: internal/plugin/logger.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func getPluginLogLevel(logger logrus.FieldLogger, pluginKey string, pluginType T
 	logLevel := os.Getenv(envName)
 
 	if logLevel == "" {
-		logger.Info("Explicitly using Info level as custom log level was not set by %q environment variable", envName)
+		logger.Infof("Explicitly using Info level as custom log level was not set by %q environment variable", envName)
```
Comment (josefkarasek) [inline internal/plugin/logger.go:55]: Unrelated to this PR.

File: pkg/execute/command.go [hunk 2]
```diff
@@ -1,96 +0,0 @@
...
-// List provides the list of allowed commands
-func (e *CommandsExecutor) List(ctx context.Context, cmdCtx CommandContext) (interactive.Message, error) {
-	cmdVerb, cmdRes := parseCmdVerb(cmdCtx.Args)
-	defer e.reportCommand(cmdVerb, cmdRes, cmdCtx.Conversation.CommandOrigin, cmdCtx.Platform)
-
-	enabledKubectls, err := e.getEnabledKubectlExecutorsInChannel(cmdCtx.Conversation.ExecutorBindings)
-	if err != nil {
-		return interactive.Message{}, fmt.Errorf("while rendering enabled executors: %s", err.Error())
-	}
-	return respond(enabledKubectls, cmdCtx, humanReadableCommandListName), nil
```
Comment (josefkarasek) [inline pkg/execute/command.go:67]: The only place where `overrideCommand` is not `nil`.

File: pkg/execute/exec.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	buf := new(bytes.Buffer)
+	w := tabwriter.NewWriter(buf, 5, 0, 1, ' ', 0)
+	fmt.Fprintln(w, "EXECUTOR\tENABLED")
+	for _, b := range bindings {
+		executor, ok := e.cfg.Executors[b]
+		if !ok {
+			continue
+		}
+		enabled := executor.Kubectl.Enabled
+		if len(executor.Plugins) > 0 {
```
Comment (josefkarasek) [inline pkg/execute/exec.go:72]: This is mutually exclusive. Is there a case where under one group plugins and non-plugin executors would be defined?

File: pkg/execute/source.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// TabularOutput sorts source groups by key and returns a printable table
+func (e *SourceExecutor) TabularOutput(bindings []string) string {
+	buf := new(bytes.Buffer)
+	w := tabwriter.NewWriter(buf, 5, 0, 1, ' ', 0)
+	fmt.Fprintln(w, "SOURCE\tENABLED\tDISPLAY NAME")
+	sort.Strings(bindings)
+	for _, k := range bindings {
+		s := e.cfg.Sources[k]
+		if len(s.Plugins) > 0 {
```
Comment (josefkarasek) [inline pkg/execute/source.go:68]: This is mutually exclusive. Is there a case where under one group plugins and non-plugin sources would be defined?

File: pkg/execute/exec.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	e.log.Debug("List executors")
+	return respond(e.TabularOutput(cmdCtx.Conversation.ExecutorBindings), cmdCtx), nil
+}
+
+// TabularOutput sorts executor groups by key and returns a printable table
+func (e *ExecExecutor) TabularOutput(bindings []string) string {
+	sort.Strings(bindings)
+	buf := new(bytes.Buffer)
+	w := tabwriter.NewWriter(buf, 5, 0, 1, ' ', 0)
+	fmt.Fprintln(w, "EXECUTOR\tENABLED")
```
Comment (josefkarasek) [inline pkg/execute/exec.go:65]: Should we add field `DisplayName` to config.Executors?

**Comment by huseyinbabal** [comment]
@josefkarasek could you please add description about how to test those changes locally?

**Comment by huseyinbabal** [comment]
@josefkarasek the configuration file should be as follows.
```
plugins:
  cacheDir: "/tmp/plugins"
  repositories:
    botkube:
      url: http://host.k3d.internal:3000/botkube.yaml

executors:
  'plugin-based':
    botkube/helm:
      enabled: true
      config: {}
  'kubectl-allow-all':
    kubectl:
      enabled: true
      namespaces:
        include:
          - ".*"
      commands:
        verbs: [ "get" ]
        resources: [ "deployments" ]
  'kubectl-not-bound-to-any-channel':
    kubectl:
      enabled: true
      namespaces:
        include:
          - ".*"
      commands:
        verbs: [ "port-forward" ]
        resources: [ "deployments" ]

sources:
  'k8s-pod-create-events':
    displayName: "k8s events for CREATE pod"
    kubernetes:
      namespaces:
        include:
          - botkube
          - test
      events:
        - create
      resources:
        - name: v1/pods

communications:
  'default-group':
    socketSlack:
      appToken: xapp-***
      botToken: xoxb-***
      channels:
        default:
          bindings:
            executors:
            - plugin-based
            - kubectl-allow-all
            - kubectl-not-bound-to-any-channel
            sources:
            - k8s-pod-create-events
          name: botkube
      enabled: true
      notification:
        type: short

rbac:
  create: true
  rules:
    - apiGroups: ["*"]
      resources: ["*"]
      verbs: ["get", "watch", "list", "create", "delete"]
```

File: pkg/execute/exec.go (hunk 5 above)
Comment (huseyinbabal) [reply pkg/execute/exec.go]: I think this is out of context, executors does not have display name from the beginning

File: pkg/execute/source.go (hunk 4 above)
Comment (huseyinbabal) [reply pkg/execute/source.go:71]: @mszostok has better context, but old notation should be gradually deprecated, and old notation have already their configurations in different level.

**Comment by huseyinbabal** [comment]
@josefkarasek I couldn't find the exact location, but below command might be confusing, maybe better to say `list commands` is deprecated, please use as follows.
Screen Shot 2023-01-02 at 17 25 19

**Review by huseyinbabal — APPROVED** [review]
LGTM with minor comments, I also added comment for testing, you can update your PR with correct values yaml

**Pull request merged**